test/SFSmartStoreTestSuite.js

```diff
@@ -1847,7 +1847,7 @@ SmartStoreTestSuite.prototype.tryAlterSoupWithSpec = function(reIndexData) {
             QUnit.equal(cursor.currentPageOrderedEntries.length, 1, "check number of rows returned");
             return self.closeCursor(cursor);
         })
-        .done(function(param) { 
```

Comment:
As a result the next call (get soup index specs) was run in parallel with this one (alter soup) and the test would fail because the old index specs would be retrieved.
I thought I might have uncovered some bug with alter soup with spec on iOS, it took me forever to realize two threads were running in parallel.